File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Drawing.Drawing2D;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Repository;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // create images for "Hatch Styles" ImageComboBoxEdit            
            CreateHatchStyleItems();


            // create images for "Custom Colors" ImageComboBoxEdit
            CreateCustomColorsItems();

            // in-place editing
            CreatedGridControlSource();

        }

        private void CreatedGridControlSource()
        {
            BindingList<Test> source = new BindingList<Test>();
            source.Add(new Test() { Style = HatchStyle.DarkDownwardDiagonal, CustomColor = Color.Red });
            source.Add(new Test() { Style = HatchStyle.DarkVertical, CustomColor = Color.Yellow });
            source.Add(new Test() { Style = HatchStyle.Percent10, CustomColor = Color.Brown });
            source.Add(new Test() { Style = HatchStyle.Trellis, CustomColor = Color.White });

            gridControl1.DataSource = source;
            gridControl1.ForceInitialize();

            RepositoryItemImageComboBox reHS = gridControl1.RepositoryItems.Add("ImageComboBoxEdit") as RepositoryItemImageComboBox;
            reHS.Assign(imageComboBoxEdit1.Properties);
            gridView1.Columns["Style"].ColumnEdit = reHS;


            RepositoryItemImageComboBox reColors = gridControl1.RepositoryItems.Add("ImageComboBoxEdit") as RepositoryItemImageComboBox;
            reColors.Assign(imageComboBoxEdit2.Properties);
            gridView1.Columns["CustomColor"].ColumnEdit = reColors;
        }

        private void CreateCustomColorsItems()
        {
            List<Color> customColors = new List<Color>();
            customColors.AddRange(new Color[] { Color.Red, Color.Black, Color.Yellow, Color.Green, Color.Blue, Color.Brown, Color.White });
            foreach (Color color in customColors)
            {
                imageComboBoxEdit2.Properties.Items.Add(new ImageComboBoxItem(color.ToString(), color));
            }

            ImageList imagesColors = new ImageList();
            imageComboBoxEdit2.Properties.SmallImages = imagesColors;

            foreach (ImageComboBoxItem item in imageComboBoxEdit2.Properties.Items)
            {
                int iWidth = 16;
                int iHeight = 16;
                Bitmap bmp = new Bitmap(iWidth, iHeight);
                using (Graphics g = Graphics.FromImage(bmp))
                {
                    g.DrawRectangle(new Pen(Color.Black, 2), 0, 0, iWidth, iHeight);
                    g.FillRectangle(new SolidBrush((Color)item.Value), 1, 1, iWidth - 2, iHeight - 2);

                }
                imagesColors.Images.Add(bmp);
                item.ImageIndex = imagesColors.Images.Count - 1;
            }
        }

        private void CreateHatchStyleItems()
        {
            imageComboBoxEdit1.Properties.Items.AddEnum(typeof(System.Drawing.Drawing2D.HatchStyle));

            ImageList imagesHS = new ImageList();
            imageComboBoxEdit1.Properties.SmallImages = imagesHS;

            foreach (ImageComboBoxItem item in imageComboBoxEdit1.Properties.Items)
            {
                int iWidth = 16;
                int iHeight = 16;
                Bitmap bmp = new Bitmap(iWidth, iHeight);
                using (Graphics g = Graphics.FromImage(bmp))
                {
                    g.DrawRectangle(new Pen(Color.Black, 2), 0, 0, iWidth, iHeight);
                    HatchBrush b = new HatchBrush((HatchStyle)item.Value, Color.Black, Color.White);
                    g.FillRectangle(b, 0, 0, iWidth, iHeight);

                }
                imagesHS.Images.Add(bmp);
                item.ImageIndex = imagesHS.Images.Count - 1;
            }
        }
    }

    public class Test
    {
        public HatchStyle Style { get; set; }
        public Color CustomColor { get; set; }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System.Drawing.Drawing2D
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Repository

Namespace WindowsFormsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' create images for "Hatch Styles" ImageComboBoxEdit            
			CreateHatchStyleItems()


			' create images for "Custom Colors" ImageComboBoxEdit
			CreateCustomColorsItems()

			' in-place editing
			CreatedGridControlSource()

		End Sub

		Private Sub CreatedGridControlSource()
			Dim source As New BindingList(Of Test)()
			source.Add(New Test() With {.Style = HatchStyle.DarkDownwardDiagonal, .CustomColor = Color.Red})
			source.Add(New Test() With {.Style = HatchStyle.DarkVertical, .CustomColor = Color.Yellow})
			source.Add(New Test() With {.Style = HatchStyle.Percent10, .CustomColor = Color.Brown})
			source.Add(New Test() With {.Style = HatchStyle.Trellis, .CustomColor = Color.White})

			gridControl1.DataSource = source
			gridControl1.ForceInitialize()

			Dim reHS As RepositoryItemImageComboBox = TryCast(gridControl1.RepositoryItems.Add("ImageComboBoxEdit"), RepositoryItemImageComboBox)
			reHS.Assign(imageComboBoxEdit1.Properties)
			gridView1.Columns("Style").ColumnEdit = reHS


			Dim reColors As RepositoryItemImageComboBox = TryCast(gridControl1.RepositoryItems.Add("ImageComboBoxEdit"), RepositoryItemImageComboBox)
			reColors.Assign(imageComboBoxEdit2.Properties)
			gridView1.Columns("CustomColor").ColumnEdit = reColors
		End Sub

		Private Sub CreateCustomColorsItems()
			Dim customColors As New List(Of Color)()
			customColors.AddRange(New Color() { Color.Red, Color.Black, Color.Yellow, Color.Green, Color.Blue, Color.Brown, Color.White })
			For Each color As Color In customColors
				imageComboBoxEdit2.Properties.Items.Add(New ImageComboBoxItem(color.ToString(), color))
			Next color

			Dim imagesColors As New ImageList()
			imageComboBoxEdit2.Properties.SmallImages = imagesColors

			For Each item As ImageComboBoxItem In imageComboBoxEdit2.Properties.Items
				Dim iWidth As Integer = 16
				Dim iHeight As Integer = 16
				Dim bmp As New Bitmap(iWidth, iHeight)
				Using g As Graphics = Graphics.FromImage(bmp)
					g.DrawRectangle(New Pen(Color.Black, 2), 0, 0, iWidth, iHeight)
					g.FillRectangle(New SolidBrush(CType(item.Value, Color)), 1, 1, iWidth - 2, iHeight - 2)

				End Using
				imagesColors.Images.Add(bmp)
				item.ImageIndex = imagesColors.Images.Count - 1
			Next item
		End Sub

		Private Sub CreateHatchStyleItems()
			imageComboBoxEdit1.Properties.Items.AddEnum(GetType(System.Drawing.Drawing2D.HatchStyle))

			Dim imagesHS As New ImageList()
			imageComboBoxEdit1.Properties.SmallImages = imagesHS

			For Each item As ImageComboBoxItem In imageComboBoxEdit1.Properties.Items
				Dim iWidth As Integer = 16
				Dim iHeight As Integer = 16
				Dim bmp As New Bitmap(iWidth, iHeight)
				Using g As Graphics = Graphics.FromImage(bmp)
					g.DrawRectangle(New Pen(Color.Black, 2), 0, 0, iWidth, iHeight)
					Dim b As New HatchBrush(CType(item.Value, HatchStyle), Color.Black, Color.White)
					g.FillRectangle(b, 0, 0, iWidth, iHeight)

				End Using
				imagesHS.Images.Add(bmp)
				item.ImageIndex = imagesHS.Images.Count - 1
			Next item
		End Sub
	End Class

	Public Class Test
		Private privateStyle As HatchStyle
		Public Property Style() As HatchStyle
			Get
				Return privateStyle
			End Get
			Set(ByVal value As HatchStyle)
				privateStyle = value
			End Set
		End Property
		Private privateCustomColor As Color
		Public Property CustomColor() As Color
			Get
				Return privateCustomColor
			End Get
			Set(ByVal value As Color)
				privateCustomColor = value
			End Set
		End Property
	End Class
End Namespace
